tests/runner.py

```diff
@@ -828,15 +828,15 @@ def assertTextDataContained(self, text1, text2):
     text2 = text2.replace('\r\n', '\n')
     return self.assertContained(text1, text2)
 
-  def assertContained(self, values, string, additional_info=''):
+  def assertContained(self, values, string, additional_info='', check_all=False):
     if type(values) not in [list, tuple]:
       values = [values]
     values = list(map(asstr, values))
     if callable(string):
       string = string()
-    for value in values:
-      if value in string:
-        return # success
+
+    if (all if check_all else any)(value in string for value in values):
```

Comment:
changes like these are pretty risky (if somehow they make tests pass by mistake, we'd not notice that). please verify manually if you haven't already that tests do still fail when they should

Reply:
Tested manually. When `check_all` is not `True`, it only fails the tests if and only if none of the strings in `expected_output` match.

Reply:
Thanks!